src/ClientRuntime/Microsoft.Rest.ClientRuntime/ServiceClient.cs

```diff
@@ -253,7 +269,7 @@ protected static HttpClientHandler CreateRootHandler()
         /// <summary>
         /// Gets the HttpClient used for making HTTP requests.
         /// </summary>
-        public HttpClient HttpClient { get; protected set; }
+        public virtual HttpClient HttpClient { get; protected set; }
 
```

Comment:
did you consider this issue? [http://stackoverflow.com/questions/119506/virtual-member-call-in-a-constructor](http://stackoverflow.com/questions/119506/virtual-member-call-in-a-constructor)